### PR TITLE
Add i18n documentation

### PR DESCRIPTION
**Pull request by fluiddot**
This PR adds documentation for i18n support in the README file.

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -156,6 +156,83 @@ To have the linter also _fix_ the violations run: `npm run lint:fix`.
...
+## Caveats
+- Strings that are only used in the native version, and reference a [context](https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/#disambiguation-by-context), won't be included in the localization strings files hence, they won't be translated. This is a limitation in the format of the localization strings files.
+- Localization strings files don’t support domains, so the strings extracted from plugins that are only used in the native version, will be unified in the same file, which might involve string conflicts.
+
+## Troubleshooting
+
+### A string is missing the translation
+This can be produced by several causes, check the following steps in order to identify the source:
+- Verify that the string uses the `__` i18n function or similar ([reference](https://github.com/WordPress/gutenberg/blob/trunk/packages/i18n/README.md)).
+- Verify warnings in the output when running `i18n:update` NPM command, especially the following ones:
```
Comment (mchowning) [inline README.md]: Something like "Check here" reads better to me than "Verify". This is definitely not a big deal at all, so feel free to leave it as-is if you prefer.

```suggestion
- Check for warnings in the output when running `i18n:update` NPM command, especially the following ones:
```

File: README.md [hunk 2]
```diff
@@ -156,6 +156,83 @@ To have the linter also _fix_ the violations run: `npm run lint:fix`.
...
+  - Parsing files issues (i.e. `Debug (make-pot): Could not parse file <FILE>`)
+  - Missing strings on translation files (i.e. `WARNING: The following strings are missing from translations:`)
```
Comment (mchowning) [inline README.md]: Replace "on" with "in"?
```suggestion
  - Missing strings in translation files (i.e. `WARNING: The following strings are missing from translations:`)
```

**Review by mchowning — APPROVED** [review]
This looks great @fluiddot ! Thank you!

Only left two very minor, non-blocking comments. :shipit:

File: README.md (hunk 1 above)
Comment (fluiddot) [reply README.md]: Thanks @mchowning for the suggestion, I've just applied it in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4455/commits/62970b52b835f41b5cc5c91b42876bed07b4a552.

File: README.md (hunk 2 above)
Comment (fluiddot) [reply README.md]: Thanks @mchowning for the suggestion, I've just applied it in https://github.com/wordpress-mobile/gutenberg-mobile/pull/4455/commits/65c2402026ab42e98dbdaeda2f8009d6b6f0267d.

**Pull request merged**